Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,93 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            // Author : Aditi Rupade.
+
+            //Splitting string a*b=c into a,b,c
+            string[] splitEquation = equation.Split(new Char[] { '*', '=' });
+
+            //Case1: num1 has ?
+
+            if (splitEquation[0].Contains('?'))
+            {
+                 int num2 = 0, res = 0;
+
+                //Converting String to Integer
+                Int32.TryParse(splitEquation[1], out num2);
+                Int32.TryParse(splitEquation[2], out res);
+
+               //Checking for factors of result(c)
+                if (res % num2 == 0)
+                {
+                    //calculating factor with missing number
+                    int temp = res / num2;
+                    string temp2 = temp.ToString();
+
+                    return FindIndex(splitEquation[0], temp2);
+                }
+
+            }
+
+            //Case2: num2 has ?
+
+            else if (splitEquation[1].Contains('?'))
+            {
+                //Works in the Same way as above function just values different
+                int num1 = 0, res = 0;
+
+                Int32.TryParse(splitEquation[0], out num1);
+                Int32.TryParse(splitEquation[2], out res);
+
+                if (res % num1 == 0)
+                {
+                    int temp = res / num1;
+                    string temp2 = temp.ToString();
+
+                    return FindIndex(splitEquation[1], temp2);
+                }
+
+            }
+
+            //Case3: res has ?
+
+            else
+            {
+                int num1 = 0, num2 = 0;
+
+                Int32.TryParse(splitEquation[0], out num1);
+                Int32.TryParse(splitEquation[1], out num2);
+
+                //calculating actual result
+                int temp = num1 * num2;
+                string temp2 = temp.ToString();
+
+                return FindIndex(splitEquation[2], temp2);
+            }
+            Console.WriteLine($"-1");

```

Comment:
console output should always have meaning full data/message given. Printing -1 does not give good experience to the user.